tests/Golden.hs

```diff
@@ -0,0 +1,77 @@
+-- make sure we don't accidentally change the result of any of the examples in
+-- the 'examples' folder.
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ViewPatterns #-}
+module Golden where
+
+import Control.Lens hiding (argument)
+import Control.Monad.Except
+import Control.Monad.Trans.Writer (WriterT, execWriterT, tell)
+import Data.Foldable (for_)
+import Data.Text.Lazy (Text)
+import System.FilePath (replaceExtension, takeBaseName)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (findByExtension, goldenVsString)
+import qualified Data.List as List
+import qualified Data.Map as Map
+import qualified Data.Text.Lazy as Text
+import qualified Data.Text.Lazy.Encoding as Text
+
+import Evaluator
+import Expander
+import Expander.Monad
+import ModuleName
+import Pretty
+import World
+
+
+mkGoldenTests :: IO TestTree
+mkGoldenTests = do
+  allKlisterFiles <- findByExtension [".kl"] "examples"
+  let klisterFiles :: [FilePath]
+      klisterFiles = filter (not . ("/non-examples/" `List.isInfixOf`))
+                   . filter (not . ("/failing-examples/" `List.isInfixOf`))
+                   $ allKlisterFiles
+  return $ testGroup "Golden tests"
+    [ goldenVsString testName goldenFile $ do
+        outputLines <- execWriterT $ runExamples klisterFile
+        pure . Text.encodeUtf8 $ outputLines
+    | klisterFile <- klisterFiles
+    , let testName = takeBaseName klisterFile
+    , let goldenFile = replaceExtension klisterFile ".golden"
+    ]
+
+runExamples :: FilePath -> WriterT Text IO ()
```

Comment:
`runExamples` and `expandFile` are both adapted from code in `repl/Main.hs`. I didn't bother moving it to a library function because I couldn't reuse it _exactly_ as is; I'm accumulating the results to a file rather than stdout, and I didn't bother printing the resulting core. Maybe I should?

Reply:
It's unfortunate that Haskell doesn't have an abstraction like Scheme ports, which would let us combine these better. I think it's fine to have this little bit of duplication, personally.